hail/src/main/scala/is/hail/expr/ir/functions/GenotypeFunctions.scala

```diff
@@ -10,32 +10,32 @@ import is.hail.expr.types.virtual.{TArray, TFloat64, TInt32, Type}
 object GenotypeFunctions extends RegistryFunctions {
 
   def registerAll() {
-    registerCode("gqFromPL", TArray(tv("N", "int32")), TInt32, (_: Type, _: PType) => PInt32()) { case (r, rt, (tPL: PArray, _pl: Code[Long])) =>
-      val pl = r.mb.newLocal[Long]("pl")
-      val m = r.mb.newLocal[Int]("m")
-      val m2 = r.mb.newLocal[Int]("m2")
-      val len = r.mb.newLocal[Int]("len")
-      val pli = r.mb.newLocal[Int]("pli")
-      val i = r.mb.newLocal[Int]("i")
-      Code(
-        pl := _pl,
-        m := 99,
-        m2 := 99,
-        len := tPL.loadLength(pl),
-        i := 0,
-        Code.whileLoop(i < len,
-          tPL.isElementDefined(pl, i).mux(
-            Code._empty,
-            Code._fatal[Unit]("PL cannot have missing elements.")),
-          pli := Region.loadInt(tPL.loadElement(pl, len, i)),
-          (pli < m).mux(
-            Code(m2 := m, m := pli),
-            (pli < m2).mux(
-              m2 := pli,
-              Code._empty)),
-          i := i + 1
-        ),
-        m2 - m)
+    registerPCode("gqFromPL", TArray(tv("N", "int32")), TInt32, (_: Type, _: PType) => PInt32())
+    { case (r, rt, _pl: PIndexableCode) =>
+      val code = EmitCodeBuilder.scopedCode(r.mb) { cb =>
+        val pl = _pl.memoize(cb, "plv")
+        val m = cb.memoize[Int](const(99), "m")
+        val m2 = cb.memoize[Int](const(99), "m2")
+        val i = cb.memoize[Int](const(0), "i")
+
+        cb.whileLoop(i < pl.loadLength(), {
+          val iec = pl.loadElement(cb, i)
+          cb += iec.Lmissing
+          cb += Code._fatal[Unit]("PL cannot have missing elements.")
+          cb += iec.Lpresent
+          val pli = cb.memoize[Int](iec.pc.tcode[Int], "pli")
```

Comment:
I would prefer changes like this (.tcode[Int] away from Region.loadInt) to be made in a separate PR, out of scope from the move to registerPCode.

Reply:
That is the scope of moving to registerPCode.

This change is two things:
1. Replacement of the single address `pl` variable with a `PIndexableValue` (created by `PIndexableCode.memoize`
1. Now that we have a `PIndexableValue`, use `loadElement` which returns an `IEmitCode` which encapsulates all of `isElementDefined`, `loadElement` (which only gets you the address), and `loadInt`.

Everything else is necessary for these two changes to take place, but not fundamental to what this change is.